service/src/main/java/com/commercetools/pspadapter/payone/config/PayoneConfig.java

```diff
@@ -28,20 +29,23 @@
     private final String integratorVersion;
 
 
-    public PayoneConfig(final PropertyProvider propertyProvider) {
-        subAccountId = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.PAYONE_SUBACC_ID);
-        merchantId = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.PAYONE_MERCHANT_ID);
-        portalId = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.PAYONE_PORTAL_ID);
-        mode = propertyProvider.getProperty(PropertyProvider.PAYONE_MODE).orElse(DEFAULT_PAYONE_MODE);
-        apiUrl = propertyProvider.getProperty(PropertyProvider.PAYONE_API_URL).orElse(DEFAULT_PAYONE_API_URL);
-        apiVersion = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.PAYONE_API_VERSION);
-        encoding = propertyProvider.getProperty(PropertyProvider.PAYONE_REQUEST_ENCODING).orElse(DEFAULT_PAYONE_REQUEST_ENCODING);
-        final String plainKey = propertyProvider.getMandatoryNonEmptyProperty(PropertyProvider.PAYONE_KEY);
+    public PayoneConfig(final TenantPropertyProvider tenantPropertyProvider) {
+        // read app common properties
+        apiUrl = tenantPropertyProvider.getPropertyProvider().getProperty(PropertyProvider.PAYONE_API_URL).orElse(DEFAULT_PAYONE_API_URL);
```

Comment:
actually, it is explicitly stated in line 33: _read app **common** properties_ and than below in line 42 _read **tenant specific** properties_. 

Anyway, i renamed the method and put some additional comments to avoid confusion: d01da3b

Let me know if it is still confusing.